### PR TITLE
28/rewards list

### PR DESCRIPTION
**Pull request by jashpam**

### CODE REVIEW

File: src/reducers/settings.js [hunk 1]
```diff
@@ -44,6 +46,11 @@ export default (state = INITIAL_STATE, { type, payload }) => {
         ...state,
         headerVisible: payload,
       };
+    case SET_BADGE_GROUP:
+      return {
+        ...state,
+        badgegroup: payload,
```
Comment (juancarlosfarah) [inline src/reducers/settings.js]: You should use camelCase: `badgeGroup`.

File: src/reducers/settings.js [hunk 2]
```diff
@@ -15,6 +16,7 @@ const INITIAL_STATE = {
   subSpaceId: null,
   open: false,
   headerVisible: true,
+  badgegroup: 0,
```
Comment (juancarlosfarah) [inline src/reducers/settings.js]: You should use camelCase: `badgeGroup`.

File: src/components/modes/teacher/AssignBadgeForm.js [hunk 3]
```diff
@@ -98,12 +108,14 @@ const mapStateToProps = (state, ownProps) => {
   const { t } = ownProps;
   return {
     studentOptions: state.users.content.map(({ id, name }) => ({ value: id, label: name })),
-    badgeOptions: state.badges.content.map(({ _id, label, color }) => ({
+    badgeOptions: state.badges.groups[state.settings.badgegroup].badges.map(({
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/AssignBadgeForm.js]: You should use camelCase: `badgeGroup`.

File: src/components/modes/teacher/Settings.js [hunk 4]
```diff
@@ -80,21 +82,32 @@ class Settings extends Component {
     this.saveSettings(settingsToChange);
   };
 
+  handleChangeGroup = (selectedGroup) => {
+    const { value } = selectedGroup;
+    const settingsToChange = {
+      badgegroup: value,
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/Settings.js]: You should use camelCase: `badgeGroup`.

File: src/components/modes/teacher/Settings.js [hunk 5]
```diff
@@ -80,21 +82,32 @@ class Settings extends Component {
...
       i18n,
       classes,
       open,
       settings,
+      badgeOptions,
     } = this.props;
 
     const {
       headerVisible,
+      badgegroup,
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/Settings.js]: You should use camelCase: `badgeGroup`.

File: src/components/modes/teacher/Settings.js [hunk 6]
```diff
@@ -80,21 +82,32 @@ class Settings extends Component {
...
     } = settings;
 
     const { language } = i18n;
     const selectedLanguage = langOptions.find(langOption => langOption.value === language);
+    const selectedGroup = badgeOptions.find(badgeOption => badgeOption.value === badgegroup);
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/Settings.js]: You should use camelCase: `badgeGroup`.

File: src/components/modes/teacher/Settings.js [hunk 7]
```diff
@@ -136,9 +164,13 @@ class Settings extends Component {
 Settings.propTypes = {
   classes: PropTypes.shape({}).isRequired,
   open: PropTypes.bool.isRequired,
+  badgeOptions: PropTypes.arrayOf(PropTypes.shape(
+    { label: PropTypes.string.isRequired, value: PropTypes.number.isRequired },
+  )).isRequired,
   settings: PropTypes.shape({
     headerVisible: PropTypes.bool.isRequired,
     lang: PropTypes.string.isRequired,
+    badgegroup: PropTypes.number.isRequired,
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/Settings.js]: You should use camelCase: `badgeGroup`.

File: src/components/modes/teacher/Settings.js [hunk 8]
```diff
@@ -148,13 +180,27 @@ Settings.propTypes = {
...
+            <FontAwesomeIcon color="stormgray" icon={_badges[0].icon} />
+            { ` ${t(name)}` }
+          </div>
+        ),
+        value: index,
+      })),
+    settings: {
+      lang: settings.lang,
+      headerVisible: settings.headerVisible,
+      badgegroup: settings.badgegroup,
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/Settings.js]: You should use camelCase: `badgeGroup`.

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Nice commit! I changed it to go to `develop` instead of `master` and have left you a couple minor comments. As soon as you have a look at that, I can merge it and work on deployment.

**Comment by jashpam** [comment]
Thanks @juancarlosfarah. I have changed 'badgegroup' to 'badgeGroup' in all files now.

**Comment by jashpam** [comment]
Thanks @juancarlosfarah. I have changed 'badgegroup' to 'badgeGroup' in all files now.

**Review by juancarlosfarah — APPROVED** [review]

**Pull request merged**